build.sbt

```diff
@@ -2,379 +2,402 @@ import java.io._
-        def isArtifactId(node: XmlNode, fn: String => Boolean) = node.label == "artifactId" && fn(node.text)
-        node.label == "dependency" && node.child.exists(child => isArtifactId(child, _.startsWith("scalameta_")))
-      }
-      override def transform(node: XmlNode): XmlNodeSeq = node match {
-        case e: Elem if isScalametaDependency(node) => Comment("scalameta dependency has been merged into scalahost via sbt-assembly")
-        case _ => node
-      }
-    }).transform(node).head
-  }
-) dependsOn (scalameta, testkit % Test)
-
-
-lazy val scalahostSbt = Project(
-  id   = "scalahostSbt",
-  base = file("scalahost-sbt")
-) settings (
-  publishableSettings,
-  buildInfoSettings,
-  sbt.ScriptedPlugin.scriptedSettings,
-  sbtPlugin := true,
-  bintrayRepository := "maven", // sbtPlugin overrides this to sbt-plugins
-  testQuick := {
-    // runs tests for 2.11 only, avoiding the need to publish for 2.12
-    RunSbtCommand(
-      "; plz 2.11.8 publishLocal " +
-      "; such scalahostSbt/scripted sbt-scalahost/simple211")(state.value)
-  },
-  test:= {
-    RunSbtCommand(
-      "; such publishLocal " +
-      "; such scalahostSbt/scripted")(state.value)
-  },
-  description := "sbt plugin to enable the scalahost compiler plugin",
-  moduleName := "sbt-scalahost",  // sbt convention is that plugin names start with sbt-
-  scalaVersion := "2.10.5",
-  crossScalaVersions := Seq("2.10.5"), // for some reason, scalaVersion.value does not work.
-  scriptedLaunchOpts ++= Seq(
-    "-Dplugin.version=" + version.value,
-    // .jvmopts is ignored, simulate here
-    "-XX:MaxPermSize=256m",
-    "-Xmx2g",
-    "-Xss2m"
-  ) ++ {
-    // pass along custom boot properties if specified
-    val bootProps = "sbt.boot.properties"
-    sys.props.get(bootProps).map(x => s"-D$bootProps=$x").toList
-  },
-  scriptedBufferLog := false
-) enablePlugins (BuildInfoPlugin)
-
-lazy val testkit = Project(
-  id   = "testkit",
-  base = file("scalameta/testkit")
-) settings (
-  publishableSettings,
-  libraryDependencies ++= Seq(
-    "com.lihaoyi" %% "geny" % "0.1.1",
-    // These are used to download and extract a corpus tar.gz
-    "org.rauschig" % "jarchivelib" % "0.7.1",
-    "commons-io" % "commons-io"  % "2.5"
-  ),
-  libraryDependencies += "org.scala-lang" % "scala-compiler" % scalaVersion.value % Test,
-  description := "Testing utilities for scala.meta's metaprogramming APIs"
-) dependsOn (scalameta)
-
-lazy val contrib = Project(
-  id   = "contrib",
-  base = file("scalameta/contrib")
-) settings (
-  publishableSettings,
-  description := "Utilities for scala.meta"
-) dependsOn (scalameta, testkit % Test)
-
-lazy val benchmarks = Project(
-    id = "benchmarks",
-    base = file("scalameta/benchmarks")
+lazy val ioJVM = io.jvm
+lazy val ioJS = io.js
+
+lazy val dialects = crossProject
+  .in(file("scalameta/dialects"))
+  .settings(publishableSettings, description := "Scala.meta's dialects", enableMacros)
```

Comment:
For arguments of `settings`, I'd prefer a uniform style of one-per-line, indented-with-two-spaces.